### PR TITLE
Lint fixes for sprint 1.15

### PR DESCRIPTION
**Pull request by Ash-KODES**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/chaincore/chain/config.go [hunk 1]
```diff
@@ -767,7 +767,10 @@ func (c *ConfigImpl) Update(fields map[string]string, version int64) error {
 	}
 
 	if maxTxnFee == 0 {
-		maxTxnFeeF = DefaultMaxTxnFee
+		maxTxnFee, err = currency.ParseZCN(DefaultMaxTxnFee)
+		if err != nil {
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/chaincore/chain/config.go]: Remove error check from here

File: code/go/0chain.net/smartcontract/storagesc/benchmark_setup.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func addMockAllocation(
 	eventDb *event.EventDb,
 	balances cstate.StateContextI,
 ) {
-	const mockWriePoolSize = 600000000
+	//const mockWriePoolSize = 600000000 // Lint issue - Unused variable
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/benchmark_setup.go]: Add unused lint ignore

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 3]
```diff
@@ -424,8 +424,9 @@ func getBlobbersForRequest(request allocationBlobbersRequest, edb *event.EventDb
 	}
 
 	if len(blobberIDs) < numberOfBlobbers && !isForce {
-		return nil, errors.New(fmt.Sprintf("not enough blobbers to honor the allocation : %d < %d", len(blobberIDs), numberOfBlobbers))
+		return nil, fmt.Errorf("not enough blobbers to honor the allocation: %d < %d", len(blobberIDs), numberOfBlobbers)
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: Don't modify error statement. You are missing a space here.

File: code/go/0chain.net/smartcontract/storagesc/models.go [hunk 4]
```diff
@@ -1379,7 +1381,7 @@ func (sa *StorageAllocation) replaceBlobber(blobberID string, sc *StorageSmartCo
 						var cp *challengePool
 						cp, e = sc.getChallengePool(sa.ID, balances)
 						if e != nil {
-							e = fmt.Errorf("could not get challenge pool of alloc: %s, err: %v", sa.ID, e)
+							_ = fmt.Errorf("could not get challenge pool of alloc: %s, err: %v", sa.ID, e)
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/models.go]: Add the error check with hardfork activator athena.

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go [hunk 5]
```diff
@@ -390,7 +391,7 @@ func (ssc *StorageSmartContract) stakePoolLock(t *transaction.Transaction,
 func (_ *StorageSmartContract) refreshProvider(
 	providerType spenum.Provider, providerID string, balances chainstate.StateContextI,
 ) (s stakepool.AbstractStakePool, err error) {
-	sp, err := getStakePool(providerType, providerID, balances)
+	sp, _ := getStakePool(providerType, providerID, balances)
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/stakepool.go]: Add error check with athena.

**Review by Jayashsatolia403 — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/models.go (hunk 4 above)
Comment (Ash-KODES) [reply code/go/0chain.net/smartcontract/storagesc/models.go]: This is already wrapped under apollo activator , do I need to do nesting with athena activator for error coming from `getChallengePool` ?

File: code/go/0chain.net/smartcontract/storagesc/challenge.go [hunk 6]
```diff
@@ -869,7 +870,7 @@ func selectRandomBlobber(selection challengeBlobberSelection, challengeBlobbersP
 			challengeBlobbers[i], challengeBlobbers[j] = challengeBlobbers[j], challengeBlobbers[i]
 		})
 
-		var blobbersSelected = make([]ChallengeReadyBlobber, 0, maxBlobbersSelect)
+		var blobbersSelected []ChallengeReadyBlobber
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/challenge.go]: Revert this change.

File: code/go/0chain.net/smartcontract/storagesc/models.go [hunk 7]
```diff
@@ -1377,14 +1379,20 @@ func (sa *StorageAllocation) replaceBlobber(blobberID string, sc *StorageSmartCo
...
 						var cp *challengePool
-						cp, e = sc.getChallengePool(sa.ID, balances)
-						if e != nil {
-							e = fmt.Errorf("could not get challenge pool of alloc: %s, err: %v", sa.ID, e)
-						}
-
-						e = sa.moveFromChallengePool(cp, d.ChallengePoolIntegralValue)
-						if e != nil {
-							e = fmt.Errorf("failed to move challenge pool back to write pool: %v", e)
+						demeterActError := cstate.WithActivation(balances, "demeter", func() (e error) { return },
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/models.go]: Why is before function empty? We need to add older changes in before function as it is and add new changes in after function.

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go [hunk 8]
```diff
@@ -390,7 +392,17 @@ func (ssc *StorageSmartContract) stakePoolLock(t *transaction.Transaction,
 func (_ *StorageSmartContract) refreshProvider(
 	providerType spenum.Provider, providerID string, balances chainstate.StateContextI,
 ) (s stakepool.AbstractStakePool, err error) {
-	sp, err := getStakePool(providerType, providerID, balances)
+	var sp *stakePool
```
Comment (Jayashsatolia403) [inline code/go/0chain.net/smartcontract/storagesc/stakepool.go]: Add older code in before function.

**Review by Jayashsatolia403 — CHANGES_REQUESTED** [review]

**Pull request merged**